dttools/src/path.c

```diff
@@ -462,4 +463,118 @@ char *path_which(const char *exec) {
 	}
 }
 
+char *path_concat(const char *p1, const char *p2) {
+	char *p = NULL;
+	size_t s1, s2;
+
+	assert(p1);
+	assert(p2);
+
+	s1 = strlen(p1);
+	s2 = strlen(p2);
+
+	p = malloc(sizeof(char) * (s1 + s2 + 2));
+	if(!p) {
+		fprintf(stderr, "path_concat malloc failed: %s!\n", strerror(errno));
+		return NULL;
+	}
+
+	snprintf(p, s1+s2+2, "%s/%s", p1, p2);
```

Comment:
Perhaps it would be better. As I see your code below you do seem to compute the position of '/' often, and tracking a bug of a path that looks like 'a///b//c' might be ugly.